dist/angular-semantic-ui.js

```diff
@@ -144,96 +143,38 @@ angular.module('angularify.semantic.checkbox', [])
-      "<input type=\"checkbox\">" +
-      "<label ng-click=\"click_on_checkbox()\" ng-transclude></label>" +
-      "</div>",
-    link: function(scope, element, attrs, ngModel) {
+    controller: function() {
+      var vm = this;
 
-      //
-      // set up checkbox class and type
-      //
-      if (scope.type == 'standard' || scope.type === undefined) {
-        scope.type = 'standard';
-        scope.checkbox_class = 'ui checkbox';
-      } else if (scope.type == 'slider') {
-        scope.type = 'slider';
-        scope.checkbox_class = 'ui slider checkbox';
-      } else if (scope.type == 'toggle') {
-        scope.type = 'toggle';
-        scope.checkbox_class = 'ui toggle checkbox';
-      } else {
-        scope.type = 'standard';
-        scope.checkbox_class = 'ui checkbox';
+      if(vm.checked) {
+        vm.ngModel = true;
       }
 
-      //
-      // set checkbox size
-      //
-      if (scope.size == 'large') {
-        scope.checkbox_class = scope.checkbox_class + ' large';
-      } else if (scope.size == 'huge') {
-        scope.checkbox_class = scope.checkbox_class + ' huge';
-      }
+      vm.classes = {
+        slider: vm.type == 'slider',
+        toggle: vm.type == 'toggle',
 
-      //
-      // set checked/unchecked
-      //
-      if (scope.checked == 'false' || scope.checked === undefined) {
-        scope.checked = false;
-      } else {
-        scope.checked = true;
-        element.children()[0].setAttribute('checked', '');
-      }
-
-      //
-      // check if the parameter disabled is available
-      //
-      if (scope.disabled == 'disabled') {
-        scope.checkbox_class += ' disabled';
-      }
-
-      //
-      // Click handler
-      //
-      element.bind('click', function() {
-        scope.$apply(function() {
-          if (scope.disabled === undefined) {
-            if (scope.checked === true) {
-              scope.checked = true;
-              scope.model = false;
-              element.children()[0].removeAttribute('checked');
-            } else {
-              scope.checked = true;
-              scope.model = true;
-              element.children()[0].setAttribute('checked', 'true');
-            }
-          }
-        });
-      });
-
-      //
-      // Watch for ng-model
-      //
-      scope.$watch('model', function(val) {
-        if (val === undefined)
-          return;
+        large: vm.size == 'large',
+        huge: vm.size == 'huge'
+      };
 
-        if (val === true) {
-          scope.checked = true;
-          element.children()[0].setAttribute('checked', 'true');
-        } else {
-          scope.checked = false;
-          element.children()[0].removeAttribute('checked');
-        }
-      });
-    }
+      vm.toggle = function() { vm.ngModel = !vm.ngModel }
+    },
+    controllerAs: 'vm',
+    bindToController: true,
+    require: [ 'checkbox', 'ngModel' ],
+    template: "<div class=\"ui checkbox\" ng-class=\"vm.classes\">" +
```

Comment:
Could you replace double quotes to single ones? I really dislike using backlash character :innocent: